### ISSUE TITLE
Questions Trying to get the blur on thumbnails

### ISSUE THREAD

**Issue by nath2125** [CollinHeist/TitleCardMaker#317]
### Installation

Docker - master tag

### Describe the Bug

I can see that you can add blurring to episode thumbnails when you either have played or unplayed an episode.

I've been trying to figure out how to do so but can't figure it out.

### Steps to reproduce the behavior

1. 
2.
...

### Screenshots
No Blur
![image](https://user-images.githubusercontent.com/99367069/224859578-3222745a-18e8-43ea-8e2b-ff366b31d6a0.png)

Blur:
![image](https://user-images.githubusercontent.com/99367069/224859814-f7ce4d11-245e-48d8-b762-a95f1e881f91.png)

### Series YAML

```yaml
libraries:
  TV Shows:
    path: /media/TV/
    plex_name: TV Shows
    media_server: plex
series:
  Mr. Robot (2015):
    episode_data_source: plex
    library: TV Shows
    font:
      color: '#CE3F39'
      file: /config/fonts/MR ROBOT.ttf
    seasons:
      0: specials_0.0
      1: season_1.0
      2: season_2.0
      3: season_3.0
      4: season_4.0
```

### Debug Log

No errors more question

**Comment by nath2125** [comment]
I have also noticed its happening not in the same way but with unplayed and played videos on other themes. When i click unplayed and played the thumbnail doesnt change

**Comment by nath2125** [comment]
Not sure if maybe the frequency of realtime monitoring for titlecardmaker is to low or something im not real sure how that monitoring process works.

**Comment by nath2125** [comment]
Just as another example

What is what was ment to happen:
![image](https://user-images.githubusercontent.com/99367069/224870185-8cb4d094-0809-4029-8171-3a5c923cb6d4.png)

What I have: (it was selected as played at time of card creation, I changed it too unplayed to see if it changed and it didn't)
![image](https://user-images.githubusercontent.com/99367069/224870207-ac58593c-f7e4-4026-bdf9-12bc529fac4c.png)

**Comment by nath2125** [comment]
An additional question being also is there a way to move the custom title thumbnail JPG it makes somewhere else? From my media folder where I keep my Plex content? I have put my tv.yml file up there, I'm not sure if there's a line in there that directs where they go to be saved? Thanks for taking your time to read and answer!

**Comment by CollinHeist** [comment]
This is described on the wiki for the Emby, Plex, and series options called `unwatched_style` and `watched_style` ([here](https://github.com/CollinHeist/TitleCardMaker/wiki/Series-YAML-Files#styles)). To blur unwatched episodes you need to have `unwatched_style: blur` either globally, for that series, library or template. 

---

Once that's enabled, here's a blurb from the wiki about when these cards are reload:

> Every time the Maker is run, it checks the "spoil status" of the last-loaded card to see if it needs to be updated. This means that if a spoiler-free card was loaded and then the episode was watched within Plex, the next time the Maker runs it will delete and remake the title card in a spoiler variant (unless art_all or blur_all is the specified unwatched action).

To get the immediate card remake that I have on the README you'll want to look at the [Tautulli integration](https://github.com/CollinHeist/TitleCardMaker/wiki/Tautulli-Integration)

**Comment by nath2125** [comment]
Do I need to add the update_script attribute in order for it to make immediate actions? With tautulli or is the basic information (e.g api, ip) enough for my preferences?

**Comment by nath2125** [comment]
excuse my message i didnt read it properly

**Comment by nath2125** [comment]
It talks about an update .txt and update .sh file which one am i putting into my preferences when going through the tautulli integration guide?

**Comment by CollinHeist** [comment]
That is the script - see [here](https://github.com/CollinHeist/TitleCardMaker/wiki/Tautulli-Integration#update-script), and [here](https://github.com/CollinHeist/TitleCardMaker/wiki/Tautulli-Integration#complete-example)

**Comment by nath2125** [comment]
so i finished the guide and set up the 2 notification scripts on tautulli 

This is what i put in preferences. (I have tried making an episode that has different watch and unwatched styles to unwatched and watched again, But no changes are happening instantly.

tautulli:
  url: http://192.168.10.4:8181/          # Your Tautulli URL
  api_key: # Your API key
  update_script: /tautulli/update_card.sh
  agent_name: Update TitleCardMaker

**Comment by CollinHeist** [comment]
Did you add the watched/unwatched style to the series config? What's your run command look like?

**Comment by CollinHeist** [comment]
I am also assuming the above YAML is just not formatted in the comment, it should really look like this:
```yaml
tautulli:
  url: http://192.168.10.4:8181/ # Your Tautulli URL
  api_key: xxxx # Your API key
  update_script: /tautulli/update_card.sh
  agent_name: Update TitleCardMaker
```

**Comment by nath2125** [comment]
I added watched\unwatched style to .tv config and yes my tautulli is configured with those indents I just didn't represent them in the comment. Not sure what u mean by run command? Is there a seperate run command that needs to happen ? Other then running the main card maker container? 

I can send you what that looks like in the .tv when I get home. As I'm not currently home right now maybe I did it wrong although it does change with what I put in just not instant it's only after it's noticed the changes when it's runs it's frequency check.

**Comment by nath2125** [comment]
for this example im talking about stranger things.

```
  Stranger Things (2016):
    episode_data_source: plex
    library: TV Shows
    card_type: Yozora/RetroTitleCard
    extras:
      unwatched_style: play
      watched_style: rewind
```

**Comment by CollinHeist** [comment]
> Is there a seperate run command that needs to happen ? Other then running the main card maker container?

There's no separate run command, but your docker run command has to be modified to include the Tautulli arguments. That's [the last step](https://github.com/CollinHeist/TitleCardMaker/wiki/Tautulli-Integration#docker) on the Tautulli setup instructions.

---

So the Tautulli integration will not work with Yozora's RetroTitleCard because that's not a built-in card and he's doing some logic separate from the TCM standard un/watched styles. 

For example:
```yaml
series:
  Stranger Things (2016):
    unwatched_style: blur
```
This works with the Tautulli integration since TCM seems the unwatched style associated with the series (not as an extra) and modifies the card.

**Comment by nath2125** [comment]
Ok I see il try without the extra and I've also noticed just now there is no update.txt file to be found which might be a cause for this issue.

since i have this already in the container it looks like 
![image](https://user-images.githubusercontent.com/99367069/225012760-757a64e0-a26a-4546-9579-119b46a6c551.png)

**Comment by nath2125** [comment]
Alright, seems I have had it working now, it's not instant but when I watch a video to 90% it does change it right when I exit the video, so I'm happy enough. Turns out, tautulli was running into permissions issues, not being able to use the .sh file and create the update.txt. Probably because I use unraid os not traditional docker. 

Just annoyed I can't seem to use those retro ones, comes up as an error when I take the "extra" away.

**Comment by nath2125** [comment]
Il try that now, weird question but is there a way to make it, so it can be adjusted to each user that uses plex, although I know the thumbnails I use and see are represented for that media and will be seen by everyone so not sure if you even can make it, so each person has a different thumbnail based around if it's been watched or not.

**Comment by CollinHeist** [comment]
Sadly that's not possible, Plex does not allow per-user assets like that.

**Comment by nath2125** [comment]
Another things I'm running into also is when I put things like this in.

I will only pick up the 4k library and not the tvshows one.

```
#4K Version	  
  Tulsa King (2022):
    episode_data_source: plex
    library: TV Shows
    library: 4K TV Shows
    card_type: Yozora/BarebonesTitleCard
    font:
      color: rgb(240,238,229)
      file: /config/fonts/Tulsaking.otf
      size: 200%
      kering: 170%
      interline_spacing: 60
```
So IL go and make 2 separate ones, and it will finally detect and make it for both. But I'm also running into issues where even if I do that, it doesn't detect it at all? Is there a way to reset its monitoring for that tvshow or something, so it can re discover it? Right now 4k has title card but not the normal 1080p ones?
```	  
  Tulsa King (2022):
    episode_data_source: plex
    library: TV Shows
    card_type: Yozora/BarebonesTitleCard
    font:
      color: rgb(240,238,229)
      file: /config/fonts/Tulsaking.otf
      size: 200%
      kering: 170%
      interline_spacing: 60
```
```	  
  Tulsa King (2022):
    episode_data_source: plex
    library: 4K TV Shows
    card_type: Yozora/BarebonesTitleCard
    font:
      color: rgb(240,238,229)
      file: /config/fonts/Tulsaking.otf
      size: 200%
      kering: 170%
      interline_spacing: 60
```

**Comment by CollinHeist** [comment]
Glad you got that figured out - feel free to open a new issue if you have any other problems!
